dhis-2/dhis-api/src/main/java/org/hisp/dhis/programrule/ProgramRuleAction.java

```diff
@@ -134,7 +134,7 @@ public class ProgramRuleAction extends BaseIdentifiableObject implements Metadat
    *   <li>sendmessage
    * </ul>
    */
-  private String templateUid;
+  private ProgramNotificationTemplate notificationTemplate;
```

Comment:
Will this affect Maintenance App?
They were sending a uid before, how is it working now?

Reply:
yes and we do have a ticket for that. 
https://dhis2.atlassian.net/browse/DHIS2-17534

Reply:
This is a breaking change.
If we want to proceed this way, we need to coordinate with Platform team and we cannot just leave the app broken for the time being.
Why cannot we leave this field as a `templateUid` and change the underneath logic to validate it and store the actual `ProgramNotificationTemplate` in the DB?

Reply:
this is exactly what I discussed previously and we agreed to have a foreign key instead of a validation. 
But foreign key is the right solution to go with. if its breaking change then we need to coordinate with the platform team.

Reply:
I still don't get it why we need to change the API.
That field could be annotated with @JsonIgnore, we convert the templateUid to a ProgramNotificationTemplate and everything works in the backend without changing the API